lib/Mojolicious/Guides/Testing.pod

```diff
@@ -0,0 +1,754 @@
+requests and check for HTTP transport errors. HTTP request assertions include
+C<get_ok>, C<post_ok>, etc. These assertions do not test whether the request
+was handled I<successfully>, only that the web application handled the request
+in an HTTP compliant way.
+
+You may also make HTTP requests using custom verbs (beyond C<GET>, C<POST>,
+C<PUT>, etc.) by building your own transaction object. See L</"Custom
+Transactions"> below.
+
+=head3 Using HTTP Request Assertions
+
+To post a URL-encoded form to the C</calls> endpoint of an application, we
+simply use the C<form> content type shortcut:
+
+  $t->post_ok('/calls', form => { to => '+43.55.555.5555' });
+
+which will create the following HTTP request:
+
+  POST /calls HTTP/1.1
+  Content-Length: 20
+  Content-Type: application/x-www-form-urlencoded
+
+  to=%2B43.55.555.5555
+
+The C<*_ok> HTTP request assertion methods accept the same arguments as their
+corresponding L<Mojo::UserAgent> methods (except for the callback
+argument). This allows us to set headers and build query strings for authentic
+test situations:
+
+  $t->get_ok('/internal/personnel',
+             { Authorization => "Internal secret-password" },
+             form => { q => "Professor Plum" });
+
+which generates the following request:
+
+  GET /internal/personnel?q=Professor+Plum HTTP/1.1
+  Content-Length: 0
+  Authorization: Internal secret-password
+
+The C<form> content generator (see L<Mojo::UserAgent::Transactor>) will
+generate a query string for C<GET> requests and
+C<application/x-www-form-urlencoded> or C<multipart/form-data> for POST
+requests.
+
+While these C<*_ok> assertions make the HTTP I<requests> we expect, they tell
+us little about I<how well> the application handled the request. The
+application we're testing might have returned any content-type, body, or HTTP
+status code (200, 302, 400, 404, 500, etc.) and we wouldn't know it.
+
+L<Test::Mojo> provides assertions to test almost every aspect of the HTTP
+response, including the HTTP response status code, the value of the
+C<Content-Type> header, and other arbitrary HTTP header information.
+
+=head2 HTTP Response Status Code
+
+While not technically an HTTP header, the status line is the first line in an
+HTTP response and is followed by the response headers. Testing the response
+status code is common in REST-based and other web applications that use the
+HTTP status codes to broadly indicate the type of response the server is
+returning.
+
+Testing the status code is as simple as adding the C<status_is> assertion:
+
+  $t->post_ok('/doorbell',
+              form => { action => "ring once" })
+    ->status_is(200);
+
+Along with C<status_isnt>, this will cover most needs. For more elaborate
+status code testing, you can access the status code directly:
+
+  $t->post_ok('/doorbell',
+              form => { action => "ring once" });
+  cmp_ok($t->tx->res->code, "<", 400, "status is good");
```

Comment:
I'm not sure I love comparing the status code as <400. Perhaps some other example can be dreamt up to fill this space?